src/Microsoft.Azure.WebJobs.Host/Extensions/IJobHostMetadataProvider.cs

```diff
@@ -60,5 +63,12 @@ public interface IJobHostMetadataProvider
         /// </summary>
         /// <param name="output">The <see cref="TextWriter"/> to write the graph to.</param>
         void DebugDumpGraph(TextWriter output);
+
+        /// <summary>
+        /// Gets function metadata.
+        /// </summary>
+        /// <param name="methodName">Method name.</param>
+        /// <returns></returns>
+        Task<FunctionMetadata> GetFunctionMetadata(string methodName, CancellationToken cancellationToken = default(CancellationToken));
```

Comment:
Shame that this needs to be async. Let's rename it to GetFunctionMetadataAsync though